plugins/plugin/functions

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+set -eo pipefail; [[ $DOKKU_TRACE ]] && set -x
+source "$PLUGIN_CORE_AVAILABLE_PATH/common/functions"
+
+disable_plugin() {
+  local PLUGIN="$1"
+  [[ -e $PLUGIN_CORE_AVAILABLE_PATH/$PLUGIN ]] && dokku_log_fail "Cannot disable a core plugin"
+  [[ ! -e $PLUGIN_ENABLED_PATH/$PLUGIN ]] && dokku_log_fail "Plugin already disabled"
+  [[ ! -e $PLUGIN_AVAILABLE_PATH/$PLUGIN ]] && dokku_log_fail "Plugin does not exist"
+  plugn disable $PLUGIN
+  dokku_log_info1_quiet "Plugin $PLUGIN disabled"
+}
+
+enable_plugin() {
+  local PLUGIN="$1"
+  [[ -e $PLUGIN_ENABLED_PATH/$PLUGIN ]] && dokku_log_fail "Plugin already enabled"
+  [[ ! -e $PLUGIN_AVAILABLE_PATH/$PLUGIN ]] && dokku_log_fail "Plugin does not exist"
+  plugn enable $PLUGIN
+  dokku_log_info1_quiet "Plugin $PLUGIN enabled"
+}
+
+download_plugin() {
+  local PLUGIN_GIT_URL="$1"
+  plugn install $PLUGIN_GIT_URL
+}
+
+download_and_enable_plugin() {
```

Comment:
Should this support plugin versioning? Since it's git, basically anything tree-ish should work I think?